packages/transport/http.go

```diff
@@ -39,35 +70,259 @@ func marshalMsgs(r *http.Request, msgs []set.Token) ([]byte, error) {
+		defer func() {
+			slog.Info("Closing connection", "connectionId", randomString)
+			delete(h.sockets, randomString)
+		}()
+		for {
+			message, ok := <-ch
+			if !ok {
+				log.Printf("Channel closed")
+				return
+			}
+			arr, _ := json.Marshal(message)
+			_, err := w.Write([]byte("data: " + string(arr) + "\n\n"))
 			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
+				log.Printf("Error writing to response: %v", err)
 				return
 			}
-			w.WriteHeader(200)
-			w.Write(arr) // nolint
+			flusher.Flush()
+		}
+	}
+}
+
+// handleAttacksIn24Hours handles the request for attacks in the last 24 hours
+func (h *httpTransport) handleAttacksIn24Hours() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		msgs := h.getter.GetAttacksIn24Hours()
+		arr, err := marshalMsgs(r, msgs)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
-		w.WriteHeader(http.StatusMethodNotAllowed)
-	}))
-	go http.ListenAndServe(":"+fmt.Sprintf("%d", h.port), mux) // nolint
-	slog.Info("HTTP transport listening", "port", h.port)
+		setDefaultHeaders(w)
+		w.WriteHeader(http.StatusOK)
+		_, err = w.Write(arr) // Check the error return value of w.Write
+		if err != nil {
+			// Handle the error appropriately
+			return
+		}
+	}
+}
 
+// handleCountIn24Hours handles the request for number of attacks per hour for last 24hours
+func (h *httpTransport) handleCountIn24Hours() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		msgs := h.getter.GetCountIn24Hours()
+		arr, err := marshalMsgs(r, msgs)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		setDefaultHeaders(w)
+		w.WriteHeader(http.StatusOK)
+		_, err = w.Write(arr) // Check the error return value of w.Write
+		if err != nil {
+			// Handle the error appropriately
+			return
+		}
+	}
 }
 
-func NewHTTP(config HTTPConfig) *httpTransport {
-	return &httpTransport{
-		port:   config.Port,
-		getter: config.Getter,
+// handleCountIn7Days handles the request for number of attacks per day for last 7 days
+func (h *httpTransport) handleCountIn7Days() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		msgs := h.getter.GetCountIn7Days()
+		arr, err := marshalMsgs(r, msgs)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		setDefaultHeaders(w)
+		w.WriteHeader(http.StatusOK)
+		_, err = w.Write(arr) // Check the error return value of w.Write
+		if err != nil {
+			// Handle the error appropriately
+			return
+		}
+	}
+}
+
+// handleCountIn6Monts handles the request for number of attacks per month for last 6 months
+func (h *httpTransport) handleCountIn6Monts() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		msgs := h.getter.GetCountIn6Months()
+		arr, err := marshalMsgs(r, msgs)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
```

Comment:
Is there any way to use helper function (get function as parameter?)for the handle endpoint to reduce redundancy in code?